### PR TITLE
Add tp.stack

### PR DESCRIPTION
**Pull request by parthchadha**

### CODE REVIEW

File: tripy/tests/integration/test_stack.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+class TestStack:
+    @pytest.mark.parametrize(
+        "tensor_shapes, dim",
+        [
+            ([(2, 3, 4), (2, 3, 4)], 0),
+            ([(2, 3, 4), (2, 3, 4)], 1),
+            ([(2, 3, 4), (2, 3, 4)], -1),
+            ([(2, 3, 4)], 0),
```
Comment (pranavm-nvidia) [inline tripy/tests/integration/test_stack.py:33]: Do we need to cover the cases where:
1. unsqueezing is required
2. The stack axis has different lengths in the inputs

File: tripy/tripy/frontend/trace/ops/stack.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (pranavm-nvidia) [inline tripy/tripy/frontend/trace/ops/stack.py:1]: Composite ops should go under `frontend/ops/`

File: tripy/tests/integration/test_stack.py (hunk 1 above)
Comment (parthchadha) [reply tripy/tests/integration/test_stack.py:33]: 1. `unsqueeze` is always required.
2. stack operates via adding a new dimension to all tensors and concatenating along them. You do not stack on existing axis

File: tripy/tripy/frontend/trace/ops/stack.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+@export.public_api(document_under="operations/functions")
+@constraints.dtype_info(
+    dtype_variables={
+        "T1": ["float32", "float16", "bfloat16", "float8", "int8", "int32", "int64", "bool"],
+    },
+    dtype_constraints={"tensors": "T1", constraints.RETURN_VALUE: "T1"},
+)
+def stack(tensors: List[Union["tripy.Tensor"]], dim: int = 0) -> "tripy.Tensor":
+    """
+    Stacks multiple tensors of same shape along a given dimension.
```
Comment (pranavm-nvidia) [inline tripy/tripy/frontend/trace/ops/stack.py:30]: Same shape except along the concat axis right?

File: tripy/tripy/frontend/trace/ops/stack.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    Args:
+        tensors: List of tensors of the same shape.
+        dim: The dimension to insert. Default: 0.
```
Comment (pranavm-nvidia) [inline tripy/tripy/frontend/trace/ops/stack.py:34]:
```suggestion
        dim: The dimension to insert. 
```

File: tripy/tripy/frontend/trace/ops/stack.py (hunk 3 above)
Comment (parthchadha) [reply tripy/tripy/frontend/trace/ops/stack.py:30]: No, stack always appends a dimension along which concatenation happens.

**Review by pranavm-nvidia — APPROVED** [review]

File: tripy/tripy/frontend/ops/stack.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    from tripy.frontend.trace.ops.unsqueeze import unsqueeze
+    from tripy.frontend.trace.ops.concatenate import concatenate
+
+    if not tensors:
+        raise_error(f"tp.stack expects a non-empty list of tensors, got {tensors}")
+
+    # Check if all tensors have the same rank
+    first_rank = tensors[0].rank
+    if not all(tensor.rank == first_rank for tensor in tensors):
```
Comment (pranavm-nvidia) [inline tripy/tripy/frontend/ops/stack.py:59]:
```suggestion
    if len(set(tensor.rank for tensor in tensors)) > 1:
```

**Review by yizhuoz004 — APPROVED** [review]

**Pull request merged**